### DISCUSSION TITLE
Would you like to see a VS Code "CC/Tweaked" extension?

### DISCUSSION THREAD

**Discussion by VewDev** [cc-tweaked/CC-Tweaked#1729] [Polls]
I have been playing on a modded world with my friends on a modpack a friend made, and discovered this mod.

I absolutely loved it! It is so much fun, and you can wget scripts you host online.

The only thing that I found annoying is that the built-in editor is even worse than nano (which I don't blame the mod for, it is still amazing to have this mod)

I thought that it would be very cool to be able to code in VS code and then git it into the server so you could wget.

However, there is no intellisense for "CC tweaked" in VS code.

Maybe using a custom ".cctweaked" extension or something.

I could make such an extension, would you guys use it if I made it (probs licensed AGPL)?

**Comment by Wojbie** [comment]
Most players who use VS Code this seems to be using https://marketplace.visualstudio.com/items?itemName=lemmmy.computercraft-extension-pack
In addition https://marketplace.visualstudio.com/items?itemName=jackmacwindows.craftos-pc allows you to setup remote access to ingame CC:T instances using web sockets.

**Reply by TheLarkInn** [reply]
Does this support connecting to remote server CC:T instances? This would the desire for me, so I can CC w/ multiple folks on a single server. There is VSCode LiveShare which is interesting protocol for connections into the server FS.

**Reply by Lupus590** [reply]
I don't know how the Craftos PC plugin works but I assume that it does the same thing that Cloud Catcher does. Cloud Catcher has a companion program that runs on the ComputerCraft computer and you get access to just that in-game computer.

**Reply by samuelh2005** [reply]
@TheLarkInn https://marketplace.visualstudio.com/items?itemName=jackmacwindows.craftos-pc does support remote access. It uses https://remote.craftos-pc.cc as a proxy service. https://remote.craftos-pc.cc gives you a link to a program which you run in game, the computer (in game) will use web sockets to forward the terminal (and receive user interaction) to https://remote.craftos-pc.cc. Then you connect to the computer using VS code and your CC terminal will magically appear inside VS code. Also, the extension allows you to mount the CC file system inside your VS code - this means you can mange files on the CC computer using VS code's file browser.

**Comment by piprett** [comment]
> However, there is no intellisense for "CC tweaked" in VS code.

From my experience you can get high quality intellisense by using the [`sumneko.lua`](https://github.com/LuaLS/lua-language-server) vscode extension with the CC: Tweaked addon for it.
Alternativly, there is also [Lua Language Server library for CC: Tweaked](https://github.com/nvim-computercraft/lua-ls-cc-tweaked)!
